### PR TITLE
Fully upgraded cloners may give genetic powers

### PR DESCRIPTION
**Pull request by XDTM**
:cl: XDTM
add: Fully upgrading a cloning pod may give clones genetic superpowers.
/:cl:

The functionality was there but it required a level 6 scanner for some reason.

On an unrelated note, it seems it should have a 50% chance of giving a disability to the clone if unupgraded, and it definitely does not do that right now.

### CODE REVIEW

**Comment by GunHog** [comment]
This used to be a feature at one point, I wonder what happened to it?

**Comment by ma44** [comment]
@XDTM could there be an option to not be the hulk? as being a hulk also brings disadvantages

**Comment by GunHog** [comment]
That is a good consideration. Hulk is not always a good thing, so I recommend you avoid setting that one.

**Comment by Jazaen** [comment]
There is mutadone. And when you get cloned you are literally next to genetics.

**Comment by XDTM** [comment]
This should remove hulk from the possible outcomes, although i kinda liked the idea of some guy hulking out of the pod and avenging his own death.

**Comment by Niknakflak** [comment]
Bad idea

**Comment by PKPenguin321** [comment]
now science can get all the powers in addition to mechs and bags of holding and autocloners oh boy

**Comment by YakumoChen** [comment]
Time to kill myself half a dozen times for science

**Comment by XDTM** [comment]
...science can already just build a genetics set up with a xenobio monkey and get all the powers in 5-10 minutes. 
And remember this requires a component that requires either a ghost chili or a lazarus injector, which is fairly rare, and 20% chance means you have to go through 5 clonings per power on average.

**Comment by YakumoChen** [comment]
Just think of it like ghetto orb of scrying!

EDIT: Actually don't do this because it's IC in OOC

**Comment by Jazaen** [comment]
Well, not really, seeing as most of the time genetics lab is by medical primarily. But genetics is part of science in heart (and it's nice to have two heads when CMO goes mad with power).

**Comment by XDTM** [comment]
By the way, does anyone know why cloning does not give bad mutations when unupgraded? It seems like it should, from the code.

File: code/game/machinery/cloning.dm [hunk 1]
```diff
@@ -166,8 +166,8 @@
 	if(efficiency > 2)
 		for(var/A in bad_se_blocks)
 			setblock(H.dna.struc_enzymes, A, construct_block(0,2))
-	if(efficiency > 5 && prob(20))
-		randmutg(H)
+	if(efficiency > 3 && prob(20))
+		randmutvg(H)
 	if(efficiency < 3 && prob(50))
 		var/mob/M = randmutb(H)
 		if(ismob(M))
```
Comment (GunHog) [inline code/game/machinery/cloning.dm]: I do not believe this statement will ever be true - `randmutb(H)` does not return a mob, but a 1 or a 0. Try testing it with just `randmutb(H)`.

**Comment by Cheridan** [comment]
powercreep

**Comment by Davdichan** [comment]
Minor powers at best, genetics should be able to do their job by upgrading the cloner and repeatedly beating themselves to death.

**Comment by optimumtact** [comment]
This is bad because it replaces a job with a machine.

**Comment by Kromgar** [comment]
You only get one power not all of them

**Comment by optimumtact** [comment]
That doesn't make it any better, it's reducing a whole job with players and sort of interesting mechanics into a thing that gets bolted on to another machine in a different department.

**Comment by YakumoChen** [comment]
Maybe if [INSERT JOB HERE] did their job science wouldn't have to resort to doing it for them

**Comment by optimumtact** [comment]
The geneticists job isn't to hand out random powers to randomly picked crewmembers and more importantly, science doesn't need to be made any more integral to a round than it already is.

**Comment by Razharas** [comment]
How is this a feature if it just restores the behaviour that was accepted(merged), but the code of it broke somehow eventually?
Thats a fix, people who think this is a bad feature are free to make new pr that will be actual balance change that removes this

**Comment by optimumtact** [comment]
@Razharas that argument only makes sense if it hasn't been like a billion years since that behaviour was restored.

I don't agree that the behaviour was ever good, so it's a :-1: from me.

**Comment by optimumtact** [comment]
If anything it's a balance/rebalance as it's adjusting existing values.

**Comment by optimumtact** [comment]
anyway with a headcoder disapproval you'll need to get @KorPhaeron to come in to bat for your feature for us to consider merging it.

**Comment by Jazaen** [comment]
Holy Smokes, Batman! That triplepost!

Erm, anyway, the chance is only 10%. Even when station is ruined and dead people are coming in waves, assuming 40 casualties, on average only 4 people would get a power, chosen at random from: Hulk, TK, XRay, Chameleon, Cold-res and maybe dwarfism.

**Comment by GunHog** [comment]
And it requires T4 scanners, does it not? That is quite uncommon for Science now, due to the nerfs.

**Comment by Kromgar** [comment]
Uh oranges... isn't the genetics superpower virus still a thing? Everyone said oh its no issue it doesnt replace genetics

**Comment by GunHog** [comment]
DNAide virus was removed.

**Comment by XDTM** [comment]
Yeah DNA virus does not give powers anymore.
But really, getting a power out of a cloner will happen 20% of the time, and timewise it's more efficient to do genetics with roundstart parts (since it's in the same bloody room) than going through cloning for about 20 times (and you can't get hulk out of it).  And that's **if** RnD has scanned a ghost chili or a lazarus injector.
This definitely won't steal or even cover genetics' job, but if it's still an issue, would it be better if i made it wipe DNA at every cloning so you can't stack them?

**Comment by KorPhaeron** [comment]
At least this is more creative than most rd upgrades (just quadrupling the numbers)

**Comment by Profakos** [comment]
What if instead of giving a random power, it would pick an unlocked one from a DNA disk inserted by the geneticists? Although this might give too much controllability to it...

File: code/game/machinery/cloning.dm (hunk 1 above)
Comment (Cheridan) [inline code/game/machinery/cloning.dm]: Do you have any comment on this @XDTM ?

**Comment by Cheridan** [comment]
Actually a fully upgraded cloner has autocloning anyway which is probably better than any superpower (except maybe x-ray which lets you catch antags behind walls without them ever knowing) so this probably isn't that much of a balance concern comparatively.

There's still the possible bug that gunhog brought up that needs to be addressed.

**Comment by GunHog** [comment]
> What if instead of giving a random power, it would pick an unlocked one from a DNA disk inserted by the geneticists? Although this might give too much controllability to it...

Actually, you can already do this. Scan somebody in the cloner, save his scan to a disk, pop it in the DNA modifier, and load a power SE (Add a new UI if you want to screw with him) to the disk, put it back in the cloner and upload it to the player's scan record. 

Few Geneticist players know how to do this, but you can REALLY screw over a new clone by adding all disabilities to the SE, or make the player into a new person entirely!

**Comment by Jazaen** [comment]
This also works the other way, you can steal genetics data from cloner saves, hehehe.
Anyway, this could be irritating in some cases, but the chance of Shittler McTide getting hulk is about 1/60.

**Comment by Jazaen** [comment]
Eh, so out of remaining powers pretty much only TK and maybe invisibility can be used to griff.

**Comment by Profakos** [comment]
@GunHog Oh, nice! I never noticed that before.

**Comment by PKPenguin321** [comment]
Does this properly exclude laser eyes

**Comment by XDTM** [comment]
Yes, only obtainable mutations of course

**Comment by XDTM** [comment]
All this controversy and i never fucking noticed that the cloning pods have _two_ scanners and not one
This should already work in game but doesn't for some reason, and this PR does not fix that.
I'll try to figure out the real problem.

**Pull request closed without merging**